en/peripherals/oneshot.md

```diff
@@ -0,0 +1,44 @@
+# OneShot Servos and ESCs (Motor Controllers)
+
+PX4 support OneShot 125 ESCs.
+However they are typically faster, and hence more responsive ([DShot](../peripherals/dshot.md) ESCs are even more responsive, and hence recommended).
+
+The [wiring setup](../peripherals/pwm_escs_and_servo.md) is exactly the same as for PWM; all you need to do is set some different parameters.
+
+
+## Overview
+
+OneShot is essentially a version of [PWM](../peripherals/pwm_escs_and_servo.md) that can be, in theory, up to 8 times faster.
+As a result, OneShot ESC are usually much more responsive and easier to tune than PWM ESC.
+
+Both PWM and OneShot communicate using a periodic pulse, where the _length_ of the pulse indicates the desired power level.
+For PWM the pulse length typically ranges between 1000uS (zero) and 2000uS (full power), while for OneShot 125 the pulse widths are 8 times shorter, ranging from 125us (zero power) to 250us (full power).
+
+The maximum rate at which pulses can be sent, and hence the responsiveness, depends on the width of the largest pulse.
+For PWM the theoretical maximum is close to 500 Hz while for OneShot it approaches 4 kHz.
+
+:::note
+The actual frame rate/duty cycle depends on the capabilities of the ESC.
+PWM ESC commonly have a rate between 50Hz and 490Hz.
+OneShot 125 ESCs are generally higher than this, though not necessarily approaching the theoretical maximum.
+:::
+
+There are a number of variants of the OneShot protocol, which support different rates.
+PX4 currently supports OneShot 125 only.
+
+## Setup
+
+### Wiring
+
+Wiring is exactly the same as for [PWM ESCs](../peripherals/pwm_escs_and_servo.md).
+
+### PX4 Configuration
+
+
+To enable OneShot use the following parameters:
+
+- DSHOT_CONFIG...= 0
```

Comment:
@bkueng 

This is draft docs, but you can see the idea. The most important bits are:
- Wiring same as PWM
- How you set up config - hence this question

I "think" you'll need to set up the frame rate / duty cycle and the max, min pulse lengths for the esc you use, possibly separately for MAIN and AUX buses.  But I can't find the parameters that you need to set . Can you point me to them - ie.. I don't see the RATE params: https://docs.px4.io/master/en/advanced_config/parameter_reference.html#pwm-outputs.

Reply:
Hmm I'll look into it why there're missing, but it's PWM_MAIN_RATE (IO) and PWM_AUX_RATE (FMU). Then you would use PWM_MAIN_{MIN,MAX} as normal, so e.g. to 1000,2000 (they are scaled internally).

(with the new configuration UI you can directly set it per output group: https://github.com/mavlink/qgroundcontrol/pull/9952).

Reply:
Would be good if you could!

In https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/pwm/pwm_main_params.yaml#L24 it says for main rate to set:

> Set to 400 for industry default or 1000 for high frequency ESCs. Set to 0 for Oneshot125.

If the pulse width for PWM is 2000us, the theoretical maximum rate is 500Hz (assuming infinitely small "off" cycle could be measured). So 400 makes sense, but what would 1000 mean?

On the net the range for normal PWM seems to be between 50Hz and 490Hz.

Reply:
I found it and will fix it. It's getting a bit difficult to generate a general parameter reference due to increasing board-specific parameters.

> If the pulse width for PWM is 2000us, the theoretical maximum rate is 500Hz (assuming infinitely small "off" cycle could be measured). So 400 makes sense, but what would 1000 mean?

Yes, I'm not aware of any real use for 1000.